float32 voltage_v
float32 remaining_percent